crates/biome_service/src/file_handlers/mod.rs

```diff
@@ -76,30 +76,83 @@ impl From<&Path> for DocumentFileSource {
 }
 
 impl DocumentFileSource {
-    /// Sorted array of files that are known as JSONC (JSON with comments).
-    ///
-    /// TODO: We might want to narrow the list according to
-    /// https://github.com/prettier/prettier/issues/15945#issuecomment-1895371835
-    ///
-    /// We can settle with these for now until someone raise a issue
-    pub(crate) const WELL_KNOWN_JSONC_FILES: &'static [&'static str; 15] = &[
+    // Well known json-like files that support comments and trailing commas
```

Comment:
This list should be updated: https://biomejs.dev/guides/how-biome-works/#protected-files

Maybe they should be part of the protected files section.